sklearn/_config.py

```diff
@@ -14,6 +14,7 @@
     ),
     "enable_cython_pairwise_dist": True,
     "array_api_dispatch": False,
+    "engine_provider": (),
```

Comment:
to enforce the experimental aspect of this api, we could have an additonal switch "`use_pluggable_engines`"  `False` by default that disable the `engine_provider` keyword if not forcibly set to `True` ?